services/121-service/test/registrations/update-chosen-fsp.test.ts

```diff
@@ -117,8 +129,92 @@ describe('Update chosen FSP of PA', () => {
 
     // Assert
     expect(response.statusCode).toBe(HttpStatus.OK);
-    expect(response.body.financialServiceProviderName).toBe(
+    expect(response.body.programFinancialServiceProviderConfigurationName).toBe(
       newProgramFinancialServiceProviderConfigurationName,
     );
   });
+
+  it('should succeed updating registration to a newly added FSP config of which the name is not the same as the FSP', async () => {
```

Comment:
is this in the end the only test needed to cover the original need of this task?